File: packages/openland-api/fragments/Message.ts
import gql from 'graphql-tag';
import { UserTiny } from './UserTiny';
import { UserShort } from './UserShort';
import { UserForMention } from './UserForMention';
import { UserBadge } from './UserBadge';
import { StickerFragment } from './StickerFragment';
import { RoomNano } from './RoomNano';

export const SpanFragment = gql`
    fragment SpanFragment on MessageSpan {
        offset
        length

        ... on MessageSpanUserMention {
            user {
                ...UserForMention
            }
        }
        ... on MessageSpanMultiUserMention {
            users {
                ...UserForMention
            }
        }
        ... on MessageSpanOrganizationMention {
            organization {
                ...OrganizationShort
            }
        }
        ... on MessageSpanRoomMention {
            room {
                ...RoomNano
            }
        }
        ... on MessageSpanLink {
            url
        }
        ... on MessageSpanDate {
            date
        }
    }

    ${UserForMention}
    ${RoomNano}
`;

export const DaialogListMessage = gql`
    fragment DaialogListMessage on ModernMessage {
        id
        date
        sender {
            id
            name
            firstName
        }
        senderBadge {
            ...UserBadge
        }
        message
        fallback
        ... on GeneralMessage {
            id
            overrideName
            overrideAvatar {
                uuid
                crop {
                    x
                    y
                    w
                    h
                }
            }
            attachments {
                id
                fallback
                ... on MessageAttachmentFile {
                    id
                    fileId
                    fileMetadata {
                        isImage
                        imageFormat
                    }
                }
            }
            quotedMessages {
                id
            }
        }
    }
    ${UserBadge}
`;

export const TinyMessage = gql`
    fragment TinyMessage on ModernMessage {
        id
        date
        sender {
            ...UserTiny
        }
        senderBadge {
            ...UserBadge
        }
        message
        fallback
        ... on GeneralMessage {
            id
            overrideName
            overrideAvatar {
                uuid
                crop {
                    x
                    y
                    w
                    h
                }
            }
            isMentioned
            commentsCount
            attachments {
                id
                fallback
                ... on MessageAttachmentFile {
                    id
                    fileId
                    fileMetadata {
                        isImage
                        imageFormat
                    }
                    filePreview
                }
            }
            quotedMessages {
                id
            }
        }
    }
    ${UserTiny}
    ${UserBadge}
`;

export const QuotedMessage = gql`
    fragment QuotedMessage on ModernMessage {
        id
        date
        message
        sender {
            ...UserShort
        }
        senderBadge {
            ...UserBadge
        }
        message
        fallback
        source {
            ... on MessageSourceChat {
                chat {
                    ... on PrivateRoom {
                        id
                    }
                    ... on SharedRoom {
                        id
                    }
                }
            }
        }
        spans {
            ...SpanFragment
        }

        ... on GeneralMessage {
            id
            overrideName
            overrideAvatar {
                uuid
                crop {
                    x
                    y
                    w
                    h
                }
            }
            commentsCount
            edited
            attachments {
                fallback
                ... on MessageAttachmentFile {
                    id
                    fileId
                    fileMetadata {
                        name
                        mimeType
                        size
                        isImage
                        imageWidth
                        imageHeight
                        imageFormat
                    }
                    filePreview
                }

                ... on MessageRichAttachment {
                    id
                    title
                    subTitle
                    titleLink
                    titleLinkHostname
                    text
                    icon {
                        url
                        metadata {
                            name
                            mimeType
                            size
                            isImage
                            imageWidth
                            imageHeight
                            imageFormat
                        }
                    }
                    image {
                        url
                        metadata {
                            name
                            mimeType
                            size
                            isImage
                            imageWidth
                            imageHeight
                            imageFormat
                        }
                    }
                    imageFallback {
                        photo
                        text
                    }
                    fallback
                }
            }
        }

        ... on StickerMessage {
            id
            date
            overrideName
            overrideAvatar {
                uuid
                crop {
                    x
                    y
                    w
                    h
                }
            }
            sender {
                ...UserShort
            }
            senderBadge {
                ...UserBadge
            }
            source {
                ... on MessageSourceChat {
                    chat {
                        ... on PrivateRoom {
                            id
                        }
                        ... on SharedRoom {
                            id
                        }
                    }
                }
            }
            reactions {
                user {
                    ...UserShort
                }
                reaction
            }
            sticker {
                ... on ImageSticker {
                    id
                    pack {
                        ... on StickerPack {
                            id
                            title
                        }
                    }
                    image {
                        ... on ImageRef {
                            uuid
                        }
                    }
                }
            }
        }
    }
`;

export const FullMessage = gql`
    fragment FullMessage on ModernMessage {
        id
        date
        sender {
            ...UserShort
        }
        senderBadge {
            ...UserBadge
        }
        message
        fallback
        source {
            ... on MessageSourceChat {
                chat {
                    ... on PrivateRoom {
                        id
                    }
                    ... on SharedRoom {
                        id
                        isChannel
                        membersCount
                    }
                }
            }
        }
        ... on GeneralMessage {
            id
            overrideName
            overrideAvatar {
                uuid
                crop {
                    x
                    y
                    w
                    h
                }
            }
            edited
            commentsCount
            attachments {
                fallback
                ... on MessageAttachmentFile {
                    id
                    fileId
                    fileMetadata {
                        name
                        mimeType
                        size
                        isImage
                        imageWidth
                        imageHeight
                        imageFormat
                    }
                    filePreview
                }

                ... on MessageRichAttachment {
                    id
                    title
                    subTitle
                    titleLink
                    titleLinkHostname
                    text
                    icon {
                        url
                        metadata {
                            name
                            mimeType
                            size
                            isImage
                            imageWidth
                            imageHeight
                            imageFormat
                        }
                    }
                    image {
                        url
                        metadata {
                            name
                            mimeType
                            size
                            isImage
                            imageWidth
                            imageHeight
                            imageFormat
                        }
                    }
                    imageFallback {
                        photo
                        text
                    }
                    keyboard {
                        buttons {
                            id
                            title
                            style
                            url
                        }
                    }
                    fallback
                }
            }
            quotedMessages {
                ...QuotedMessage
            }

            reactions {
                user {
                    ...UserShort
                }
                reaction
            }
        }

        spans {
            ...SpanFragment
        }
        
        ... on StickerMessage {
            id
            overrideName
            overrideAvatar {
                uuid
                crop {
                    x
                    y
                    w
                    h
                }
            }
            date
            commentsCount
            sender {
                ...UserShort
            }
            senderBadge {
                ...UserBadge
            }
            source {
                ... on MessageSourceChat {
                    chat {
                        ... on PrivateRoom {
                            id
                        }
                        ... on SharedRoom {
                            id
                        }
                    }
                }
            }
            quotedMessages {
                ...QuotedMessage
            }
            reactions {
                user {
                    ...UserShort
                }
                reaction
            }
            sticker {
                ...StickerFragment
            }
        }

        ... on ServiceMessage {
            id
            overrideName
            overrideAvatar {
                uuid
                crop {
                    x
                    y
                    w
                    h
                }
            }
            serviceMetadata {
                ... on InviteServiceMetadata {
                    users {
                        ...UserTiny
                    }
                    invitedBy {
                        ...UserTiny
                    }
                }

                ... on KickServiceMetadata {
                    user {
                        ...UserTiny
                    }
                    kickedBy {
                        ...UserTiny
                    }
                }

                ... on TitleChangeServiceMetadata {
                    title
                }

                ... on PhotoChangeServiceMetadata {
                    photo
                }

                ... on PostRespondServiceMetadata {
                    respondType
                }
            }
        }
    }
    ${UserTiny}
    ${UserShort}
    ${UserBadge}
    ${SpanFragment}
    ${QuotedMessage}
    ${StickerFragment}
`;

File: android/app/src/main/java/com/openland/react/graphql/operations/queries/Message.kt
package com.openland.react.graphql

import com.openland.spacex.*
import com.openland.spacex.gen.*
import org.json.*

internal val MessageSelector = obj(
            field("message", "message", arguments(fieldValue("messageId", refValue("messageId"))), obj(
                    field("__typename", "__typename", notNull(scalar("String"))),
                    field("source", "source", obj(
                            field("__typename", "__typename", notNull(scalar("String"))),
                            inline("MessageSourceChat", obj(
                                field("__typename", "__typename", notNull(scalar("String"))),
                                field("chat", "chat", notNull(obj(
                                        field("__typename", "__typename", notNull(scalar("String"))),
                                        inline("PrivateRoom", obj(
                                            field("__typename", "__typename", notNull(scalar("String"))),
                                            field("id", "id", notNull(scalar("ID"))),
                                            field("user", "user", notNull(obj(
                                                    field("__typename", "__typename", notNull(scalar("String"))),
                                                    field("id", "id", notNull(scalar("ID"))),
                                                    field("isBot", "isBot", notNull(scalar("Boolean"))),
                                                    field("name", "name", notNull(scalar("String")))
                                                ))),
                                            field("pinnedMessage", "pinnedMessage", obj(
                                                    field("__typename", "__typename", notNull(scalar("String"))),
                                                    field("id", "id", notNull(scalar("ID")))
                                                ))
                                        )),
                                        inline("SharedRoom", obj(
                                            field("__typename", "__typename", notNull(scalar("String"))),
                                            field("id", "id", notNull(scalar("ID"))),
                                            field("title", "title", notNull(scalar("String"))),
                                            field("isChannel", "isChannel", notNull(scalar("Boolean"))),
                                            field("membersCount", "membersCount", notNull(scalar("Int"))),
                                            field("canSendMessage", "canSendMessage", notNull(scalar("Boolean"))),
                                            field("canEdit", "canEdit", notNull(scalar("Boolean"))),
                                            field("repliesEnabled", "repliesEnabled", notNull(scalar("Boolean"))),
                                            field("pinnedMessage", "pinnedMessage", obj(
                                                    field("__typename", "__typename", notNull(scalar("String"))),
                                                    field("id", "id", notNull(scalar("ID")))
                                                )),
                                            field("role", "role", notNull(scalar("String")))
                                        ))
                                    )))
                            ))
                        )),
                    fragment("ModernMessage", FullMessageWithoutSourceSelector)
                )),
            field("comments", "comments", arguments(fieldValue("peerId", refValue("messageId"))), notNull(obj(
                    field("__typename", "__typename", notNull(scalar("String"))),
                    field("id", "id", notNull(scalar("ID"))),
                    field("subscription", "subscription", obj(
                            field("__typename", "__typename", notNull(scalar("String"))),
                            field("type", "type", scalar("String"))
                        ))
                )))
        )
val Message = object: OperationDefinition {
    override val name = "Message"
    override val kind = OperationKind.QUERY
    override val body = "query Message(\$messageId:ID!){message(messageId:\$messageId){__typename ...FullMessageWithoutSource source{__typename ... on MessageSourceChat{__typename chat{__typename ... on PrivateRoom{__typename id user{__typename id isBot name}pinnedMessage{__typename id}}... on SharedRoom{__typename id title isChannel membersCount canSendMessage canEdit repliesEnabled pinnedMessage{__typename id}role}}}}}comments(peerId:\$messageId){__typename id subscription{__typename type}}}fragment FullMessageWithoutSource on ModernMessage{__typename id seq date sender{__typename ...MessageSender}senderBadge{__typename ...UserBadge}message fallback spans{__typename ...MessageSpan}... on GeneralMessage{__typename id edited commentsCount attachments{__typename ...MessageAttachments}quotedMessages{__typename ...QuotedMessage}reactionCounters{__typename ...MessageReactionCounter}overrideAvatar{__typename uuid crop{__typename x y w h}}overrideName}... on StickerMessage{__typename id commentsCount quotedMessages{__typename ...QuotedMessage}sticker{__typename ...StickerFragment}reactionCounters{__typename ...MessageReactionCounter}overrideAvatar{__typename uuid crop{__typename x y w h}}overrideName}... on ServiceMessage{__typename id serviceMetadata{__typename ...ServiceMessageMetadata}}}fragment MessageSender on User{__typename id name photo isBot shortname inContacts primaryOrganization{__typename id name shortname}}fragment UserBadge on UserBadge{__typename id name verified}fragment MessageSpan on MessageSpan{__typename offset length ... on MessageSpanUserMention{__typename user{__typename id name}}... on MessageSpanMultiUserMention{__typename offset length}... on MessageSpanOrganizationMention{__typename organization{__typename id name}}... on MessageSpanRoomMention{__typename room{__typename ... on PrivateRoom{__typename id user{__typename id name}}... on SharedRoom{__typename id title isPremium}}}... on MessageSpanLink{__typename url}... on MessageSpanDate{__typename date}}fragment MessageAttachments on ModernMessageAttachment{__typename fallback ... on MessageAttachmentFile{__typename id fileId fileMetadata{__typename name mimeType size isImage imageWidth imageHeight imageFormat}filePreview}... on MessageRichAttachment{__typename id title subTitle titleLink titleLinkHostname text featuredIcon icon{__typename url metadata{__typename name mimeType size isImage imageWidth imageHeight imageFormat}}image{__typename url metadata{__typename name mimeType size isImage imageWidth imageHeight imageFormat}}socialImage{__typename url metadata{__typename name mimeType size isImage imageWidth imageHeight imageFormat}}imageFallback{__typename photo text}keyboard{__typename buttons{__typename id title style url}}fallback}... on MessageAttachmentPurchase{__typename id purchase{__typename id state amount}fallback}}fragment QuotedMessage on ModernMessage{__typename id date message sender{__typename ...MessageSender}senderBadge{__typename ...UserBadge}fallback source{__typename ... on MessageSourceChat{__typename chat{__typename ... on PrivateRoom{__typename id}... on SharedRoom{__typename id isChannel membersCount}}}}spans{__typename ...MessageSpan}... on GeneralMessage{__typename id edited commentsCount attachments{__typename ...MessageAttachments}}... on StickerMessage{__typename id sticker{__typename ...StickerFragment}}}fragment StickerFragment on Sticker{__typename ... on ImageSticker{__typename id pack{__typename id title}image{__typename uuid}}}fragment MessageReactionCounter on ReactionCounter{__typename reaction count setByMe}fragment ServiceMessageMetadata on ServiceMetadata{__typename ... on InviteServiceMetadata{__typename users{__typename id}invitedBy{__typename id}}... on KickServiceMetadata{__typename user{__typename id}kickedBy{__typename id}}... on TitleChangeServiceMetadata{__typename title}... on PhotoChangeServiceMetadata{__typename photo}... on PostRespondServiceMetadata{__typename respondType}}"
    override val selector = MessageSelector
}